### PR TITLE
Flex editor updates

### PR DESCRIPTION
**Pull request by br648**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Editor updates covering:

1. Booking Rules
2. Location Groups
3. Locations
4. Location shapes

### CODE REVIEW

**Review by binh-dam-ibigroup — COMMENTED** [review]
Could you advise on persisting the flex field after computing `presetFlex`, please?

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
See https://github.com/ibi-group/datatools-server/pull/439#pullrequestreview-866518230 and resolve failing tests, please.

File: src/main/java/com/conveyal/datatools/manager/models/FeedSource.java [hunk 1]
```diff
@@ -701,4 +712,11 @@ public FeedTransformRules getRulesForRetrievalMethod(FeedRetrievalMethod retriev
             .flatMap(Collection::stream)
             .collect(Collectors.toList());
     }
+
+    /**
+     * Set the flex value according to the feed version loaded.
+     */
+    public void presetFlex(FeedVersion feedVersion) {
+        flex = feedVersion.feedLoadResult.isGTFSFlex();
```
[the remark this answers is not in the document]
Comment (br648) [reply src/main/java/com/conveyal/datatools/manager/models/FeedSource.java]: Partially addressed with the inclusion of `flexUIFeaturesEnabled`.

File: src/main/java/com/conveyal/datatools/manager/models/FeedVersion.java [hunk 2]
```diff
@@ -274,6 +274,12 @@ public void load(MonitorableJob.Status status, boolean isNewVersion) {
             assignGtfsFileAttributes(gtfsFile);
             String gtfsFilePath = gtfsFile.getPath();
             this.feedLoadResult = GTFS.load(gtfsFilePath, DataManager.GTFS_DATA_SOURCE);
+
+
+            FeedSource flexSetFeedSource = this.parentFeedSource();
+            flexSetFeedSource.flex = this.feedLoadResult.isGTFSFlex();
```
Comment (br648) [inline src/main/java/com/conveyal/datatools/manager/models/FeedVersion.java]: @miles-grant-ibigroup is feedLoadResult guaranteed not to be null at this point? I had to add a check for this in `FeedSource#presetFlex`.

Comment (miles-grant-ibigroup) [reply src/main/java/com/conveyal/datatools/manager/models/FeedVersion.java]: I've made a corresponding gtfs-lib commit that always returns true or false https://github.com/conveyal/gtfs-lib/pull/335/commits/e4d5e611f14aceefc06676756c8f70e4c2180473

Comment (br648) [reply src/main/java/com/conveyal/datatools/manager/models/FeedVersion.java]: @miles-grant-ibigroup It is still possible for `feedLoadResult` to be null and therefore fail when a call is made to `isGTFSFlex()` method.

Comment (miles-grant-ibigroup) [reply src/main/java/com/conveyal/datatools/manager/models/FeedVersion.java]: Thanks for looking into and fixing that!

File: src/main/java/com/conveyal/datatools/manager/models/FeedSource.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (br648) [reply src/main/java/com/conveyal/datatools/manager/models/FeedSource.java]: Addressed in 49db0e1689980059c9742bd25f17d641cb064099

**Comment by br648** [comment]
@binh-dam-ibigroup reviews on this PR have stalled some what. I've now addressed this comment from you: https://github.com/ibi-group/datatools-server/pull/439#discussion_r794757400, so I think this is ready for another review.

File: src/main/java/com/conveyal/datatools/manager/models/FeedSource.java [hunk 3]
```diff
@@ -775,14 +774,18 @@ public <T extends FeedTransformation> List<T> getActiveTransformations(FeedVersi
...
+        boolean isFlexFeedVersion = false;
+        if (feedVersion.feedLoadResult != null) isFlexFeedVersion = feedVersion.feedLoadResult.isGTFSFlex();
+        if (!flex && isFlexFeedVersion) {
+            // If the feed version is flex (and the feed source was not flex) permanently enable flex.
+            flex = flexUIFeaturesEnabled = true;
+            Persistence.feedSources.updateField(this.id, "flexUIFeaturesEnabled", true);
+            Persistence.feedSources.updateField(this.id, "flex", true);
+        }
     }
+
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/models/FeedSource.java]: Remove extra blank line.

**Review by binh-dam-ibigroup — APPROVED** [review]
Things seem to work for the most part, great job!

**Review by miles-grant-ibigroup — APPROVED** [review]

**Pull request merged**